src/Domain/Campaign.php

```diff
@@ -91,19 +105,25 @@ public function __construct(
         ?string $status,
         string $name,
         string $currencyCode,
+        bool $isRegularGiving,
+        ?\DateTimeImmutable $regularGivingCollectionEnd,
     ) {
         $this->createdNow();
         $this->campaignFundings = new ArrayCollection();
-
         $this->charity = $charity;
-        $this->startDate = $startDate;
-        $this->endDate = $endDate;
         $this->salesforceId = $sfId->value;
-        $this->isMatched = $isMatched;
-        $this->ready = $ready;
-        $this->status = $status;
-        $this->name = $name;
-        $this->currencyCode = $currencyCode;
+
+        $this->updateFromSfPull(
+            currencyCode: $currencyCode,
+            status: $status,
+            endDate: $endDate,
+            isMatched: $isMatched,
+            name: $name,
+            startDate: $startDate,
+            ready: $ready,
+            isRegularGiving: $isRegularGiving,
+            regularGivingCollectionEnd: $regularGivingCollectionEnd,
+        );
```

Comment:
delegating to updateFromSfPull as that now does some sanity checks on the data that are equally relevant at construction time. Also to help make sure that any future changes to fields pulled from SF get reflected both in new campaigns and updated campaigns.